lib/_stream_writable.js

```diff
@@ -432,11 +439,15 @@ function clearBuffer(stream, state) {
     holder.entry = entry;
 
     var count = 0;
+    var allBuffers = true;
     while (entry) {
       buffer[count] = entry;
+      if (!entry.isBuf)
```

Comment:
I tried something similar (continuing the rest of the loop without any conditional) and did not see any difference in performance.